File: 3-react-go-microservice/api/fizzBuzz.go
package main

type FizzBuzzResult struct {
	Input int `json:"input"`
	Output string `json:"output"`
}

func fizzBuzz(start int, end int) (result []FizzBuzzResult) {

	fizz := "Fizz"
    buzz := "Buzz"
    empty := ""

    if end >= start {
    for i := start; i <= end; i++ {
        	if i % 15 == 0 {
                result = append(result, FizzBuzzResult{i, fizz + buzz})
            } else if i % 3 == 0 {
                result = append(result, FizzBuzzResult{i, fizz})
            } else if i % 5 == 0 {
                result = append(result, FizzBuzzResult{i, buzz})
            } else {
                result = append(result, FizzBuzzResult{i, empty})
            }
        }
    }
	return result
}

File: 2-react/src/fizzBuzz.ts
export type FizzBuzzOutput = 'Fizz' | 'Buzz' | 'FizzBuzz' | ''

export type FizzBuzzResult = {
  input: number
  output: FizzBuzzOutput
}

export const fizzBuzz = (start: number, end: number): FizzBuzzResult[] => {
  let calculation: FizzBuzzResult[] | any[] = [];
  if (end >= start) {
    for (let i = start; i <= end; i++) {
      if(i % 15 === 0) {
        calculation = [ ...calculation, { input: i, output: 'FizzBuzz'}];
      }
      else if(i % 3 === 0) {
        calculation = [ ...calculation, { input: i, output: 'Fizz'}];
      }
      else if(i % 5 === 0) {
        calculation = [ ...calculation, { input: i, output: 'Buzz'}];
      }
      else {
        calculation = [ ...calculation, { input: i, output: ''}];
      }
    }
  }
  return calculation;
}